### ISSUE TITLE
Initial temperature from XDMF for transient simulations

### ISSUE THREAD

**Issue by KulaginVladimir** [festim-dev/FESTIM#659]
According to the [discussion](https://festim.discourse.group/t/how-to-set-initial-temperature-distribution-from-xdmf/17), currently, there is no possibility to set the `initial_value` for `HeatTransferProblem` from a XDMF file.

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin. So, I came up with the following modifications for the `create_functions` method of the `HeatTransferProblem` class:

```python
        if self.transient:
            if isinstance(self.initial_value, festim.TemperatureFromXDMF):
                f.XDMFFile(self.initial_value.filename).read_checkpoint(
                    self.T, self.initial_value.label, -1
                )
                self.T_n.assign(self.T)
            else:
                ccode_T_ini = sp.printing.ccode(self.initial_value)
                self.initial_value = f.Expression(ccode_T_ini, degree=2, t=0)
                self.T_n.assign(f.interpolate(self.initial_value, V))
``` 
Thus, one could provide `F.TemperatureFromXDMF(filename = '...', label = '...')` as an input. Or, maybe, it should be done the other way?

**Comment by RemDelaporteMathurin** [comment]
@KulaginVladimir I don't think we want to use the `TemperatureFromXDMF` class for this as we would like users to keep using the `InitialCondition` class.

Instead of `if isinstance(self.initial_value, festim.TemperatureFromXDMF)` we could simply have:

```python
if isinstance(self.initial_value, str):
    if self.initial_value.endswith(".xdmf"):
        ....
        # read from XDFMFile and 
```

Does that make sense?

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin
> I don't think we want to use the TemperatureFromXDMF class for this as we would like users to keep using the InitialCondition class.

Sorry, but I didn't really get it 😄. Do you suggest to pass `F.InitialCondition` directly to `F.HeatTransferProblem`? If so, the temperature field can be read from XDMF based on `value` (string), `label` and `time_step` attributes of `F.InitialCondition`, but, I suppose, it would require to check that `initial_value` of `F.HeatTransferProblem` is of `InitialCondition` class (similar to my previous comment). Maybe. I am wrong.

The `label` is reqired to read from XDMF 👇 , isn't it?
```python
if isinstance(self.initial_value, str):
    if self.initial_value.endswith(".xdmf"):
        ....
        # read from XDFMFile and 
``` 

On the another hand, do you mean to create a possibility to set `InitialCondition` for temperature in `F.Simulation.initial_conditions`? I suppose, It would require some additional changes made in generic_simulation.py, since currently no initial conditions are passed to `HeatTransferProblem` in the `initialise()` method:
```python
# Define temperature
if isinstance(self.T, festim.HeatTransferProblem):
    self.T.create_functions(self.materials, self.mesh, self.dt)
elif isinstance(self.T, festim.Temperature):
    self.T.create_functions(self.mesh)
```

**Comment by RemDelaporteMathurin** [comment]
Yes you are correct we also need `label` and `timestep`, forgot about this!

So I still think we can use the `InitialCondition` class as is since it has `label` and `timestep`

We just need to adapt these lines in `HeatTransferProblem` like you said:
https://github.com/festim-dev/FESTIM/blob/6364d4607aaae5e0e15fe9dd3daadd6c576b477d/festim/temperature/temperature_solver.py#L77-L80

We just have to do exactly like we do for concentrations:

https://github.com/festim-dev/FESTIM/blob/6364d4607aaae5e0e15fe9dd3daadd6c576b477d/festim/concentration/concentration.py#L44-L50


The usage would be:

```python
my_temp = F.HeatTransferProblem(
    ...,
    initial_value=F.InitialCondition("T", "filename.xdmf", "label", timestep=-1)
)
```
Note that specifying `"T"` here is redundant but required...

Does this make more sense?

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin, looks nice!

**Comment by RemDelaporteMathurin** [comment]
Feel free to open a PR!

**Comment by RemDelaporteMathurin** [comment]
Fixed in #662
